### PR TITLE
Update the way state is saved, fix menubar autohide

### PR DESCRIPTION
**Pull request by sallar**
This PR fixes:

- Windows data loss #354 
- Auto hide menubar option was being shown to wrong OS #282

### CODE REVIEW

File: src/main/app.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { app, ipcMain as ipc } from 'electron';
 import pify from 'pify';
 import log from 'electron-log';
-import { throttle } from 'lodash';
+import debounce from 'lodash/debounce';
```
Comment (perry-mitchell) [inline src/main/app.js]: I'd almost recommend [this one](https://www.npmjs.com/package/debounce), so that you wouldn't have to use lodash at all..

Comment (perry-mitchell) [reply src/main/app.js]: But if it works it works..

**Comment by perry-mitchell** [comment]
I restarted the travis build too, as it failed..

File: src/main/app.js (hunk 1 above)
Comment (sallar) [reply src/main/app.js]: @perry-mitchell if we didnt use lodash anywhere else I would use this, maybe I'll remove lodash dep in the future as its really heavy

**Pull request merged**